### PR TITLE
fix(command-dev): warn when background functions are not supported

### PR DESCRIPTION
**Pull request by erezrokah**
**- Summary**

Fixes https://github.com/netlify/cli/issues/1553
Also related to https://github.com/netlify/cli/issues/958

**- Test plan**

Tested locally with an account that supports background functions, and another one that doesn't

See example warning:
image

**- Description for the changelog**

Warn when background functions are not supported when `netlify dev` is invoked

**- A picture of a cute animal (not mandatory but encouraged)**

image

### CODE REVIEW

File: src/commands/dev/index.js [hunk 1]
```diff
@@ -198,7 +199,15 @@ class DevCommand extends Command {
       exit(1)
     }
 
-    await startFunctionsServer({ settings, site, log, warn, errorExit, siteInfo: this.netlify.cachedConfig.siteInfo })
```
Comment (erezrokah) [inline src/commands/dev/index.js:201]: `siteInfo` was passed only for the sake of getting `ssl_url`

File: src/lib/account.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+const dotProp = require('dot-prop')
+
+const supportsBooleanCapability = (account, capability) => {
```
Comment (erezrokah) [inline src/lib/account.js:3]: A non boolean capability is build minuets (for example)

File: src/utils/dev.js [hunk 3]
```diff
@@ -2,28 +2,26 @@ const process = require('process')
...
-const getSiteData = async ({ api, site, failAndExit }) => {
-  try {
-    const siteData = await api.getSite({ siteId: site.id })
-    return siteData
-  } catch (error) {
-    failAndExit(
-      `Failed retrieving site data for site ${chalk.yellow(site.id)}: ${error.message}. ${ERROR_CALL_TO_ACTION}`,
-    )
+const validateSiteInfo = ({ site, siteInfo, failAndExit }) => {
+  if (isEmpty(siteInfo)) {
```
Comment (erezrokah) [inline src/utils/dev.js:16]: An empty object is returned on error from `@netlify/config`

File: src/utils/dev.js [hunk 4]
```diff
@@ -38,50 +36,64 @@ const getAddons = async ({ api, site, failAndExit }) => {
...
-    return siteData.build_settings.env
+const getSiteEnv = ({ siteInfo }) => {
+  if (siteInfo.build_settings && siteInfo.build_settings.env) {
+    return siteInfo.build_settings.env
   }
   return {}
 }
 
-const getSiteInformation = async ({ flags = {}, api, site, warn, error: failAndExit }) => {
+const getSiteInformation = async ({ flags = {}, api, site, warn, error: failAndExit, siteInfo }) => {
```
Comment (erezrokah) [inline src/utils/dev.js:68]: `siteInfo` is passed from `@netlify/config` to avoid calling `getSite` twice

File: src/utils/serve-functions.js [hunk 5]
```diff
@@ -339,9 +354,8 @@ const createFormSubmissionHandler = function (siteInfo) {
           ...fields,
           ...Object.entries(files).reduce((prev, [name, { url }]) => ({ ...prev, [name]: url }), {}),
         }).map(([key, val]) => ({ title: capitalize(key), name: key, value: val })),
-        site_url: siteInfo.ssl_url,
+        site_url: siteUrl,
       },
-      site: siteInfo,
```
Comment (erezrokah) [inline src/utils/serve-functions.js:344]: Passing `siteInfo` to the form submission function handler is redundant.

File: src/utils/serve-functions.js [hunk 6]
```diff
@@ -410,48 +424,54 @@ const getBuildFunction = ({ functionBuilder, log }) => {
   }
 }
 
-const startFunctionsServer = async ({ settings, site, log, warn, errorExit, siteInfo }) => {
```
Comment (erezrokah) [inline src/utils/serve-functions.js:413]: Broke this down into multiple functions

File: src/lib/account.js (hunk 2 above)
Comment (ehmicky) [reply src/lib/account.js:3]: Should this be added as a code comment?

**Review by ehmicky — APPROVED** [review]

**Pull request merged**